kernel/Kconfig

```diff
@@ -600,6 +600,14 @@ config SYSTEM_WORKQUEUE_NO_YIELD
 	  cooperative and a sequence of work items is expected to complete
 	  without yielding.
 
+config SYSTEM_WORKQUEUE_NO_BLOCK
+	bool "Select whether system work queue enforces non-blocking work items"
+	help
```

Comment:
Maybe `default y if ASSERT` or something similar?  This is a cheap check with clear value, probably wants to be on any time CONFIG_ASSERT=y